kotlinx-coroutines-core/common/src/Builders.common.kt

```diff
@@ -219,11 +219,15 @@ private const val RESUMED = 2
 internal class DispatchedCoroutine<in T>(
     context: CoroutineContext,
     uCont: Continuation<T>
-) : ScopeCoroutine<T>(context, uCont) {
+) : ScopeCoroutine<T>(context, uCont, true) {
     // this is copy-and-paste of a decision state machine inside AbstractionContinuation
     // todo: we may some-how abstract it via inline class
     private val _decision = atomic(UNDECIDED)
 
+    protected override fun initParentForNativeUndispatchedCoroutine() {
```

Comment:
These changes don't seem to be related to the new memory model, but rather copy-pasted from `native-mt`.

Could you please elaborate on whether they are necessary and why?